# HEAL
- Machine learning-based genome analysis and risk prediction model.

## supported file type
- VCF files

## requirements
- python3
- pandas
- numpy
- scikit-learn
- scipy


## How to run
1. Prepare cohort VCF file of exome or whole genome sequencing.
2. Annotate VCF file
   a. using ANNOVAR
   b. with allele frequency information, deleteriousness score
4. Preprosess annotated genotype data
5. Run the machine learning model. The model outputs
   a. Disease gene lists
   b. Genetic risk prediction model

## Citation
Please cite the following paper
- ......

<img width="200" alt="snyderlab_logo" src="https://github.com/pirocv/HEAL/assets/51925146/0c17a201-9642-4da3-9457-5ff83ddb9a1b">
<img width="100" alt="hp_riken" src="https://github.com/pirocv/HEAL/assets/51925146/b37c836b-1a0e-4a2b-aca5-39baf220e4ea">
<img width="100" alt="cgi_logo" src="https://github.com/pirocv/HEAL/assets/51925146/d487a395-6741-4093-b515-ea078c685333">

